### PR TITLE
Vector mosaic: support granule filtering and picking granules from a repository

### PR DESCRIPTION
**Pull request by aaime**
Extend the vector mosaic store to support filtering granule stores, and picking them from a repository

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [ ] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

File: modules/unsupported/vector-mosaic/src/main/java/org/geotools/vectormosaic/GranuleStoreFinderImpl.java [hunk 1]
```diff
@@ -24,20 +24,25 @@
...
 
 /** Class used to find the DataStore for a VectorMosaicGranule. */
 public class GranuleStoreFinderImpl extends GranuleStoreFinder {
     static final Logger LOGGER = Logging.getLogger(GranuleStoreFinderImpl.class);
     protected final String preferredSPI;
+    protected final Repository repository;
 
     /**
      * Constructor that accepts a nullable preferred SPI.
      *
```
Comment (turingtestfail) [inline modules/unsupported/vector-mosaic/src/main/java/org/geotools/vectormosaic/GranuleStoreFinderImpl.java:40]: Add Repository to doc?

File: modules/unsupported/vector-mosaic/src/main/java/org/geotools/vectormosaic/VectorMosaicFeatureSource.java [hunk 2]
```diff
@@ -66,6 +67,9 @@
 /** FeatureSource for a vector mosaic. */
 public class VectorMosaicFeatureSource extends ContentFeatureSource {
     static final Logger LOGGER = Logging.getLogger(VectorMosaicFeatureSource.class);
+
```
Comment (turingtestfail) [inline modules/unsupported/vector-mosaic/src/main/java/org/geotools/vectormosaic/VectorMosaicFeatureSource.java:70]: Add a note about what pattern this regex looks for with store names?

**Review by turingtestfail — CHANGES_REQUESTED** [review]
Adding the option to use the repo with a filter seems like a big enough change to warrant some user documentation changes?

File: modules/unsupported/vector-mosaic/src/main/java/org/geotools/vectormosaic/VectorMosaicFeatureSource.java (hunk 2 above)
Comment (aaime) [reply modules/unsupported/vector-mosaic/src/main/java/org/geotools/vectormosaic/VectorMosaicFeatureSource.java:70]: Yes, done, and added the same in the GeoServer user docs.

**Comment by aaime** [comment]
Documentation updates here: https://github.com/geoserver/geoserver/pull/7393

**Review by turingtestfail — APPROVED** [review]

**Pull request merged**